process/block/cutoff/blockProcessingCutoffHandler.go

```diff
@@ -0,0 +1,130 @@
+package cutoff
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/multiversx/mx-chain-core-go/core/check"
+	"github.com/multiversx/mx-chain-core-go/data"
+	"github.com/multiversx/mx-chain-go/common"
+	"github.com/multiversx/mx-chain-go/config"
+	logger "github.com/multiversx/mx-chain-logger-go"
+)
+
+var log = logger.GetOrCreate("process/block/cutoff")
+
+type blockProcessingCutoffHandler struct {
+	config config.BlockProcessingCutoffConfig
+}
+
+// NewBlockProcessingCutoffHandler will return a new instance of blockProcessingCutoffHandler
+func NewBlockProcessingCutoffHandler(cfg config.BlockProcessingCutoffConfig) (*blockProcessingCutoffHandler, error) {
+	err := checkConfig(cfg)
+	if err != nil {
+		return nil, err
+	}
+
+	return &blockProcessingCutoffHandler{
+		config: cfg,
+	}, nil
+}
+
+// HandlePauseCutoff will pause the processing if the required coordinates are met
+func (b *blockProcessingCutoffHandler) HandlePauseCutoff(header data.HeaderHandler) {
+	shouldSkip := !b.config.Enabled ||
+		check.IfNil(header) ||
+		b.config.Mode != common.BlockProcessingCutoffModePause
+	if shouldSkip {
+		return
+	}
+
+	blockingCutoffFunction := func(printArgs ...interface{}) error {
+		log.Info("cutting off the block processing. The node will not advance", printArgs...)
+		go func() {
+			for {
+				time.Sleep(time.Minute)
+				log.Info("node is in block processing cut-off mode", printArgs...)
+			}
+		}()
+		neverEndingChannel := make(chan struct{})
+		<-neverEndingChannel
+
+		return nil // should not reach this point
+	}
+
+	_ = b.handleCutoffIfCoordinatesAreMet(header, blockingCutoffFunction)
+	// should never reach this point
+}
+
+// HandleProcessErrorCutoff will return error if the processing the block at the required coordinates
+func (b *blockProcessingCutoffHandler) HandleProcessErrorCutoff(header data.HeaderHandler) error {
+	shouldSkip := !b.config.Enabled ||
+		check.IfNil(header) ||
+		b.config.Mode != common.BlockProcessingCutoffModeProcessError
+	if shouldSkip {
+		return nil
+	}
+
+	return b.handleCutoffIfCoordinatesAreMet(header, func(printArgs ...interface{}) error {
+		log.Info("block processing cutoff - return err", printArgs...)
+		return errProcess
+	})
+}
+
+func (b *blockProcessingCutoffHandler) handleCutoffIfCoordinatesAreMet(header data.HeaderHandler, cutOffFunction func(printArgs ...interface{}) error) error {
+	value := b.config.Value
+
+	switch common.BlockProcessingCutoffTrigger(b.config.CutoffTrigger) {
```

Comment:
this switch is executed each time. 
Let's refactor and use the single switch on the config parsing function. Please do the following:
1. have 3 uint64/uint32 values set to the max uint values. `stopRound` `stopNonce` `stopEpoch`
2. set the provided value from the config to the correct variable
3. add if conditions for all stop variables, which one will be met, the function will return true, as the condition is met.

Reply:
refatored